File: app/schemas/Asset.py
from __future__ import annotations
from typing import Optional
from pydantic import BaseModel, Field
from app.domain.enums import AssetType, Currency

class AssetCreate(BaseModel):
    ticker: str = Field(min_length=1, max_length=20)
    asset_type: AssetType
    current_price: float
    currency: Currency = Currency.EUR

class AssetRead(BaseModel):
    id: int
    ticker: str
    asset_type: AssetType
    current_price: float
    currency: Currency

class AssetUpdate(BaseModel):
    ticker: Optional[str] = Field(default=None, min_length=1, max_length=20)
    asset_type: Optional[AssetType] = None
    current_price: Optional[float] = None
    currency: Optional[Currency] = None

File: frontend/src/app/models/asset.ts
export enum AssetType { ETF = 'ETF', STOCK = 'STOCK', CRYPTO = 'CRYPTO' }
export enum Currency { USD = 'USD', EUR = 'EUR' }

export interface AssetRead {
  id: number;
  ticker: string;
  asset_type: AssetType;
  current_price: number;
  currency: Currency;
}

export interface AssetCreate {
  ticker: string;
  asset_type: AssetType;
  current_price: number;
  currency: Currency;
}

export type AssetUpdate = Partial<AssetCreate>;
